lib/recognizer/hal.ex

```diff
@@ -0,0 +1,37 @@
+defmodule Recognizer.Hal do
+  @moduledoc """
+  A simple API client for the Hal API
+
+  """
+
+  def update_newsletter(user) do
+    req = HTTPoison.get!(build_url("/accounts/newsletter/interests"))
```

Comment:
Please don't nest function calls in Elixir:

```suggestion
    req = "/accounts/newsletter/interests" |> build_url() |> HTTPoison.get!()
```?